src/includes/Services/CollectCmsDataService.php

```diff
@@ -0,0 +1,227 @@
+use Alma\Woocommerce\Exceptions\AlmaException;
+use Alma\Woocommerce\Exceptions\AlmaInvalidSignatureException;
+use Alma\Woocommerce\Helpers\SecurityHelper;
+use Alma\Woocommerce\WcProxy\FunctionsProxy;
+use Alma\Woocommerce\WcProxy\OptionProxy;
+use Alma\Woocommerce\WcProxy\ThemeProxy;
+
+/**
+ * CollectCmsDataService
+ */
+class CollectCmsDataService {
+
+	/**
+	 * Alma Settings.
+	 *
+	 * @var AlmaSettings
+	 */
+	protected $alma_settings;
+
+	/**
+	 * Security Helper.
+	 *
+	 * @var SecurityHelper
+	 */
+	protected $security_helper;
+
+	/**
+	 * Alma Logger.
+	 *
+	 * @var AlmaLogger
+	 */
+	protected $alma_logger;
+
+	/**
+	 * Payload Formatter.
+	 *
+	 * @var PayloadFormatter
+	 */
+	protected $payload_formatter;
+
+	/**
+	 * Option Proxy.
+	 *
+	 * @var OptionProxy
+	 */
+	protected $option_proxy;
+
+	/**
+	 * Theme Proxy.
+	 *
+	 * @var ThemeProxy
+	 */
+	protected $theme_proxy;
+
+	/**
+	 * Functions Proxy.
+	 *
+	 * @var FunctionsProxy
+	 */
+	protected $functions_proxy;
+
+	/**
+	 * CollectCmsDataService constructor.
+	 *
+	 * @param AlmaSettings     $alma_settings Alma Settings.
+	 * @param AlmaLogger       $alma_logger Alma Logger.
+	 * @param PayloadFormatter $payload_formatter Payload Formatter.
+	 * @param SecurityHelper   $security_helper Security Helper.
+	 * @param OptionProxy      $option_proxy Option Proxy.
+	 * @param ThemeProxy       $theme_proxy Theme Proxy.
+	 * @param FunctionsProxy   $functions_proxy Functions Proxy.
+	 */
+	public function __construct(
+		$alma_settings = null,
+		$alma_logger = null,
+		$payload_formatter = null,
+		$security_helper = null,
+		$option_proxy = null,
+		$theme_proxy = null,
+		$functions_proxy = null
+	) {
+		$this->alma_settings     = isset( $alma_settings ) ? $alma_settings : new AlmaSettings();
+		$this->alma_logger       = isset( $alma_logger ) ? $alma_logger : new AlmaLogger();
+		$this->payload_formatter = isset( $payload_formatter ) ? $payload_formatter : new PayloadFormatter();
+		$this->security_helper   = isset( $security_helper ) ? $security_helper : ( new SecurityHelperBuilder() )->get_instance();
+		$this->option_proxy      = isset( $option_proxy ) ? $option_proxy : new OptionProxy();
+		$this->theme_proxy       = isset( $theme_proxy ) ? $theme_proxy : new ThemeProxy();
+		$this->functions_proxy   = isset( $functions_proxy ) ? $functions_proxy : new FunctionsProxy();
+	}
+
+	/**
+	 * Handle collect cms data
+	 *
+	 * @return void
+	 */
+	public function handle_collect_cms_data() {
+		if ( ! array_key_exists( 'HTTP_X_ALMA_SIGNATURE', $_SERVER ) ) {
+			$this->alma_logger->error( 'Header key X-Alma-Signature doesn\'t exist' );
+			$this->functions_proxy->send_http_response( array( 'error' => 'Header key X-Alma-Signature doesn\'t exist' ), 403 );
+			return;
+		}
+
+		try {
+			$this->security_helper->validate_collect_data_signature( $this->alma_settings->get_active_merchant_id(), $this->alma_settings->get_active_api_key(), $_SERVER['HTTP_X_ALMA_SIGNATURE'] );
```

Comment:
lines are too long, can you return to line ?

Reply:
Done